### ISSUE TITLE
Invalid dependency for Ubuntu 19.04

### ISSUE THREAD

**Issue by GJRTimmer** [TryGhost/Ghost-Desktop#343]
In Ubuntu 19.04 `libgconf2-4` is renamed to `libgconf-2-4` because of the dummy package `libgconf2-4` has been removed from the repository for `Ubuntu 19.04 Disco Dingo`, could you please fix, Ghost-Desktop debian package.

Thanks in advance.

Current Install Error:
```bash
Selecting previously unselected package ghost.
(Reading database ... 124398 files and directories currently installed.)
Preparing to unpack ghost-desktop-1.7.0-debian.deb ...
Unpacking ghost (1.7.0-1) ...
dpkg: dependency problems prevent configuration of ghost:
 ghost depends on libgconf2-4; however:
  Package libgconf2-4 is not installed.

dpkg: error processing package ghost (--install):
 dependency problems - leaving unconfigured
Processing triggers for gnome-menus (3.32.0-1ubuntu1) ...
Processing triggers for desktop-file-utils (0.23-4ubuntu1) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Errors were encountered while processing:
 ghost

```

**Comment by axelitus** [comment]
I have tried manually installing `libgconf-2-4`, but as @GJRTimmer said it's not working anymore in Ubuntu 19.04.

I don't think the issue is with Ghost, but with electron-installer-debian as seen in this issue: https://github.com/electron-userland/electron-installer-debian/issues/141

A commit to fixing this has been issued on March 10, but I don't know if it has been yet released. If so, the dependency should be updated to accommodate this situation and I don't know how much it needs to change because of it.

**Comment by axelitus** [comment]
In fact, I have found out that there are at least 3 dependency definitions of `libgconf2-4` across the `node_modules` folder that are not suited for this recent change.

As a workaround, I have successfully replaced this with the more suitable `libgconf-2-4 | libgconf2-4` string to take into account the renaming of the library (and the missing dummy package).

Here are the search results of `libgconf2-4` in the `node_modules` folder:

```
../Ghost-Desktop/node_modules/electron-forge/node_modules/electron-installer-debian/src/dependencies.js:
    4  
    5  const dependencyMap = {
    6:   gconf: 'libgconf-2-4 | libgconf2-4',
    7    glib2: 'libglib2.0-bin',
    8    gtk2: 'libgtk2.0-0',

../Ghost-Desktop/node_modules/electron-forge/node_modules/electron-installer-snap/src/yaml.js:
   28  
   29  const DEPENDENCY_MAP = {
   30:   gconf: 'libgconf2-4',
   31    gtk2: 'desktop-gtk2',
   32    gtk3: 'desktop-gtk3',

../Ghost-Desktop/node_modules/electron-installer-debian/src/dependencies.js:
   31      return [
   32        trashDependencies,
   33:       'libgconf2-4',
   34        'libgtk2.0-0',
   35        'libnotify4',

../Ghost-Desktop/node_modules/electron-installer-snap/resources/snapcraft.yaml:
   29      plugin: dump
   30      stage-packages:
   31:     - libgconf2-4
   32      - libnotify4
   33      - libnss3
```

Just replace the latter 3 entries with the given string and run `npm run make`. The _proper_ installer will be in the `electron-out/make` folder.

**Comment by axelitus** [comment]
Digging a bit more around this I found that maybe the package that needs fix the issue is `ember-electron` which has some old dependencies of the offending modules:

"electron-installer-debian": "0.8.1" vs. latest "electron-installer-debian": "1.2.0"
"electron-installer-snap": "2.0.1" vs. latest "electron-installer-snap": "3.2.0"

So maybe an issue should be opened there.

**Comment by jgrocha** [comment]
On Raspbian buster, there is no `libgconf2-4` dummy package available. This dependency is also invalid for Raspberry (with `Raspbian GNU/Linux 10 (buster)`).

**Comment by justathoughtor2** [comment]
Hey all, updating the dependencies seems to have resolved the issue locally for me. I opened #344 to have this deployed to the master branch. Once it's live in master, we just need a release deployed to create a new .deb file compatible with 19.04. I've been able to electron:make the app and install the resulting .deb file successfully. Ghost Desktop is now running beautifully on my 19.04 installation.

Does anyone know if we're allowed to upload .deb files to these threads? I don't want to violate security protocols for this repository, but someone might be in need of an installable .deb for 19.04. I'm just sitting on it till the pull request is merged and a new release is pushed.

**Comment by axelitus** [comment]
Sadly, this project seems to be no longer active (the latest commit was on Jan 1 just to update the copyrights year...)

**Comment by justathoughtor2** [comment]
Yeah, it's odd. Seems like they redirected focus exclusively to the web
platform. I can understand that, though. They've added some unique features
like AMP support and automatic deployment of PWAs, so I can't really
complain.
